test/e2e/egress_firewall.go

```diff
@@ -395,193 +396,14 @@ spec:
-			if err == nil {
-				framework.Failf("Was able to curl node %s from container %s on node %s with no allow rule for egress firewall", node.Name, srcPodName, serverNodeInfo.name)
-			}
-		}
-
-		ginkgo.By("Should NOT be able to reach each secondary hostIP via node selector")
-		for _, address := range toCurlSecondaryNodeIPAddresses.List() {
-			if !IsIPv6Cluster(f.ClientSet) && utilnet.IsIPv6String(address) || IsIPv6Cluster(f.ClientSet) && !utilnet.IsIPv6String(address) {
-				continue
-			}
-			path := fmt.Sprintf("http://%s:%d/hostname", address, hostNetworkPort)
-			_, err = e2ekubectl.RunKubectl(f.Namespace.Name, "exec", srcPodName, testContainerFlag, "--", "curl", "-g", "--max-time", "2", path)
-			if err == nil {
-				framework.Failf("Was able to curl node %s from container %s on nodeIP %s with no allow rule for egress firewall", address, srcPodName, serverNodeInfo.name)
-			}
-		}
-
-		ginkgo.By("Apply label to nodes " + f.Namespace.Name + ":" + labelMatch)
-		patch := struct {
-			Metadata map[string]interface{} `json:"metadata"`
-		}{
-			Metadata: map[string]interface{}{
-				"labels": map[string]string{f.Namespace.Name: labelMatch},
-			},
-		}
-		for _, node := range nodes.Items {
-			patchData, err := json.Marshal(&patch)
-			gomega.Expect(err).NotTo(gomega.HaveOccurred())
-			f.ClientSet.CoreV1().Nodes().Patch(context.TODO(), node.Name, types.MergePatchType, patchData, metav1.PatchOptions{})
-		}
-
-		ginkgo.By("Should be able to reach each host networked pod via node selector")
-		for _, node := range nodes.Items {
-			path := fmt.Sprintf("http://%s:%d/hostname", node.Status.Addresses[0].Address, hostNetworkPort)
-			_, err = e2ekubectl.RunKubectl(f.Namespace.Name, "exec", srcPodName, testContainerFlag, "--", "curl", "-g", "--max-time", "2", path)
-			if err != nil {
-				framework.Failf("Failed to curl node %s from container %s on node %s: %v", node.Name, srcPodName, serverNodeInfo.name, err)
-			}
-		}
-
-		ginkgo.By("Should be able to reach secondary hostIP via node selector")
-		for _, address := range toCurlSecondaryNodeIPAddresses.List() {
-			if !IsIPv6Cluster(f.ClientSet) && utilnet.IsIPv6String(address) || IsIPv6Cluster(f.ClientSet) && !utilnet.IsIPv6String(address) {
-				continue
-			}
-			path := fmt.Sprintf("http://%s:%d/hostname", address, hostNetworkPort)
-			_, err = e2ekubectl.RunKubectl(f.Namespace.Name, "exec", srcPodName, testContainerFlag, "--", "curl", "-g", "--max-time", "2", path)
-			if err != nil {
-				framework.Failf("Failed to curl node %s from container %s on nodeIP %s", address, srcPodName, serverNodeInfo.name)
-			}
-		}
-	})
-
-	ginkgo.It("Should validate the egress firewall DNS does not deadlock when adding many dnsNames", func() {
-		var egressFirewallConfig = fmt.Sprintf(`kind: EgressFirewall
+	table.DescribeTable("Should validate the egress firewall policy functionality against cluster nodes by using node selector",
+		func(checkDeadlock bool) {
```

Comment:
it might be good either in a comment or in ginkgo steps explain what kind of deadlock this test is testing for. Is it something specific in the code? or what lock are you trying to exercise. Otherwise someone else looking at this test wont know what we are testing.

Reply:
i tried to do that under `if checkDeadlock` comment, any ideas on what else to add?

Reply:
I meant specify which mutexes you are trying to exercise here in the code

Reply:
ah, so as this is e2e, and not a unit test, I am not trying to reproduce a specific deadlock situation, but rather a "tricky e2e scenario" where multiple things can go wrong. Maybe the test name is not the best, but I just tried to adapt what was already there.